src/diffusers/models/transformers/sana_transformer_2d.py

```diff
@@ -0,0 +1,702 @@
+
+
+# Modified from diffusers.models.autoencoders.autoencoder_dc.GLUMBConv
+@maybe_allow_in_graph
+class SanaGLUMBConv(nn.Module):
+    def __init__(self, in_channels: int, out_channels: int, expand_ratio: float = 2.5) -> None:
+        super().__init__()
+
+        hidden_channels = int(expand_ratio * in_channels)
+
+        self.nonlinearity = nn.SiLU()
+
+        self.conv_inverted = nn.Conv2d(in_channels, hidden_channels * 2, 1, 1, 0)
+        self.conv_depth = nn.Conv2d(hidden_channels * 2, hidden_channels * 2, 3, 1, 1, groups=hidden_channels * 2)
+        self.conv_point = nn.Conv2d(hidden_channels, out_channels, 1, 1, 0, bias=False)
+
+    def forward(self, hidden_states: torch.Tensor, HW: Optional[tuple[int]] = None) -> torch.Tensor:
+        B, N, C = hidden_states.shape
+        if HW is None:
+            H = W = int(N**0.5)
+        else:
+            H, W = HW
+
+        hidden_states = hidden_states.reshape(B, H, W, C).permute(0, 3, 1, 2)
+
+        hidden_states = self.conv_inverted(hidden_states)
+        hidden_states = self.nonlinearity(hidden_states)
+
+        hidden_states = self.conv_depth(hidden_states)
+        hidden_states, gate = torch.chunk(hidden_states, 2, dim=1)
+        hidden_states = hidden_states * self.nonlinearity(gate)
+
+        hidden_states = self.conv_point(hidden_states)
+        hidden_states = hidden_states.reshape(B, C, N).permute(0, 2, 1)
+
+        return hidden_states
+
+
+# Modified from diffusers.models.attention.BasicTransformerBlock
+@maybe_allow_in_graph
+class SanaLinearTransformerBlock(nn.Module):
+    r"""
+    A Transformer block following the Linear Transformer architecture, introduced in Sana
+
+    Reference: https://arxiv.org/abs/2410.10629
+
+    Parameters:
+        dim (`int`): The number of channels in the input and output.
+        num_attention_heads (`int`): The number of heads to use for multi-head attention.
+        attention_head_dim (`int`): The number of channels in each head.
+        context_pre_only (`bool`): Boolean to determine if we should add some blocks associated with the
+            processing of `context` conditions.
+    """
+
+    def __init__(
+        self,
+        dim: int,
+        num_attention_heads: int,
+        attention_head_dim: int,
+        dropout=0.0,
+        num_cross_attention_heads: Optional[int] = None,
+        cross_attention_head_dim: Optional[int] = None,
+        cross_attention_dim: Optional[int] = None,
+        activation_fn: tuple = ("silu", "silu", None),
+        num_embeds_ada_norm: Optional[int] = None,
+        attention_bias: bool = False,
+        upcast_attention: bool = False,
+        norm_type: str = "ada_norm_single",
+        norm_elementwise_affine: bool = False,
+        norm_eps: float = 1e-6,
+        attention_out_bias: bool = True,
+        use_pe: bool = False,
+        num_positional_embeddings: Optional[int] = None,
+        expand_ratio: float = 2.5,
```

Comment:
Could we default all these parameters to the exact ones Sana transformer requires, such taht creating `SanaLinearTransformerBlock()` object without passing any parameters creates the expected block? Note that we will still pass all the parameters but it serves as a helpful reference for what branches of code will be unused that we can remove.